src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -201,19 +204,18 @@ public bool Delete([NotNull] Document document, ConcurrencyControl concurrencyCo
         /// </summary>
         /// <param name="id">The ID to use when creating or getting the document</param>
         /// <returns>The instantiated document, or <c>null</c> if it does not exist</returns>
+        /// <exception cref="CouchbaseLiteException">Thrown with <see cref="C4ErrorCode.NotOpen"/>
+        /// if this method is called after the collection is closed</exception>
         [CanBeNull]
         public Document GetDocument([NotNull] string id)
         {
-            CheckCollectionValid();
-            var doc = new Document(this, id);
-            if (!doc.Exists || doc.IsDeleted) {
-                doc.Dispose();
-                WriteLog.To.Database.V(Tag, "Requested existing document {0}, but it doesn't exist",
-                    new SecureLogString(id, LogMessageSensitivity.PotentiallyInsecure));
+            CBDebug.MustNotBeNull(WriteLog.To.Database, Tag, nameof(id), id);
+            if (id.Equals("")) {
```

Comment:
Careful, this seems like a violation of responsibility.  If the ID is invalid, then the Document constructor should say so, not here.